product_sticker/tests/common.py

```diff
@@ -0,0 +1,113 @@
+import base64
+import io
+
+from PIL import Image
+
+from odoo.tests import common, tagged
+
+
+@tagged("post_install", "-at_install")
+class ProductStickerCommon(common.TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        pa_model = cls.env["product.attribute"]
+        pav_model = cls.env["product.attribute.value"]
+        ps_model = cls.env["product.sticker"]
+        pt_model = cls.env["product.template"]
+        # No Variant Attribute
+        cls.at_p = pa_model.create({"name": "Platform", "create_variant": "always"})
+        cls.at_p_l = pav_model.create({"attribute_id": cls.at_p.id, "name": "Linux"})
+        cls.at_p_w = pav_model.create({"attribute_id": cls.at_p.id, "name": "Windows"})
+        # Create Variant Attribute
+        cls.at_l = pa_model.create({"name": "License", "create_variant": "no_variant"})
+        cls.at_l_o = pav_model.create(
+            {"attribute_id": cls.at_l.id, "name": "One-time Payment"}
+        )
+        cls.at_l_s = pav_model.create(
+            {"attribute_id": cls.at_l.id, "name": "Subscription"}
+        )
+        cls.at_l_f = pav_model.create({"attribute_id": cls.at_l.id, "name": "Freemium"})
+        # Products
+        cls.product_as400 = pt_model.create(
+            {
+                "name": "Amazing Software 400 - Diskette",
+                "default_code": "AS400",
+                "attribute_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "attribute_id": cls.at_p.id,  # Platform
+                            "value_ids": [
+                                (
+                                    6,
+                                    0,
+                                    [
+                                        cls.at_p_l.id,  # Linux
+                                        cls.at_p_w.id,  # Windows
+                                    ],
+                                )
+                            ],
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "attribute_id": cls.at_l.id,  # License
+                            "value_ids": [
+                                (
+                                    6,
+                                    0,
+                                    [
+                                        cls.at_l_o.id,  # One-time Payment
+                                        cls.at_l_s.id,  # Subscription
```

Comment:
If you need comments to explain your variables... that means that you should use better variable names! 🤣 

```suggestion
                            "attribute_id": cls.att_platform.id,
                            "value_ids": [
                                (
                                    6,
                                    0,
                                    [
                                        cls.attval_linux.id,
                                        cls.attval_windows.id,
                                    ],
                                )
                            ],
                        },
                    ),
                    (
                        0,
                        0,
                        {
                            "attribute_id": cls.att_license.id,
                            "value_ids": [
                                (
                                    6,
                                    0,
                                    [
                                        cls.attval_onetime.id,
                                        cls.attval_subscription.id, 
```

similar across the file.